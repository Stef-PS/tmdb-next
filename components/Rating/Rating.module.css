.rating {
  text-align: center;
  letter-spacing: +2px
}

.star {
  font-size: var(--font-size-m);
  color: var(--color-warning-light);
  -webkit-text-stroke: 1px var(--color-warning-light);
}

.starEmpty {
  font-size: var(--font-size-m);
  color: var(--color-white);
  -webkit-text-stroke: 1px var(--color-warning-mid);
}
