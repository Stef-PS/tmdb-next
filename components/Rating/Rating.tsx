import styles from './Rating.module.css'

interface RatingProps {
  rate: number,
}

const Rating = ({ rate }: RatingProps) => {
  const onNumber = Math.round(rate / 2)
  return (
    <div className={styles.rating}>
      {[...Array(5)].map((_star, index) => {        
        const starClass = index < onNumber ? styles.star : styles.starEmpty
        return (
          <span key={index} className={starClass}>&#9733;</span>        
        );
      })}
    </div>
  )
}

export default Rating
