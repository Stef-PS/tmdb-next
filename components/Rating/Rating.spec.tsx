import { render } from '@testing-library/react'
import Rating from './Rating'

describe('Rating', () => {
  it('should render Rating unchanged', () => {
    const { container } = render(<Rating rate={3} />)
    expect(container).toMatchSnapshot()
  })

  it('should display the right number of filled stars', () => {
    for (let i = 0; i <= 10; i++) {
      const { container } = render(<Rating rate={i} />)
      const filledStars = container.getElementsByClassName('star')?.length ?? 0
      expect(filledStars).toBe(Math.round(i / 2))
    }
  })
})

