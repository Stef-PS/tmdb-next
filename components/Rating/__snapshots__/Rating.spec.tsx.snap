// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating should render Rating unchanged 1`] = `
<div>
  <div
    class="rating"
  >
    <span
      class="star"
    >
      ★
    </span>
    <span
      class="star"
    >
      ★
    </span>
    <span
      class="starEmpty"
    >
      ★
    </span>
    <span
      class="starEmpty"
    >
      ★
    </span>
    <span
      class="starEmpty"
    >
      ★
    </span>
  </div>
</div>
`;
